### ISSUE TITLE
cross validation based on probability

### ISSUE THREAD

**Issue by jmschrei** [jmschrei/pomegranate#174]
We need a cross-validation metric, likely a thin wrapper over sklearn, which returns probability not accuracy.

**Comment by willsilvano** [comment]
@jmschrei Any news about that?

**Comment by willsilvano** [comment]
@jmschrei  Is there any other way to verify the accuracy of the Bayesian network?

**Comment by jmschrei** [comment]
Howdy. Sorry, I've been consumed by paper writing and trying to pass my general exam recently, and so have had no time to add things into pomegranate :( I'm glad that you're asking questions though, so that I know what things to focus on when I get more time.

You can test the accuracy of a Bayesian network is a similar manner to any estimator. Let's say that you have some incomplete data matrix that you want to fill in using the inference step of a pre-trained Bayesian network.  You can evaluate how well this network can fill in missing entries in the same way you evaluate a classic machine learning model-- hold out some entries in the matrix as "validation data" and use the remainder as the input to the model. You can then use the `predict` method to get the predicted labels, and use your favorite metric to evaluate the predictions.

The above approach works generally, but if you have some specific variable (or set of variables) that you always want to predict, make sure to only hold those out, rather than selecting entries randomly.

Another approach is to calculate the log probability of a subset of samples (not entries this time, full samples) that the model was not trained on. This is basically like the "validation log probability", and you might want to optimize this in the same way you optimize validation accuracy when you do hyper-parameter selection. You can imagine iteratively building models on, say, 80% of the data set, seeing how well they perform on the held out 20% (using `model.log_probability(X).sum()`) and chosing the model that performs best there.

Does that help?

**Comment by willsilvano** [comment]
Hi @jmschrei !!
Your tips were great!
I was able to measure the accuracy of my network.
But I'm having bad results. :( 
The results are only good in one of my classes. 
The main variable that I need to evaluate is having a very poor ranking result.
